internal/vm/vsock.go

```diff
@@ -277,8 +282,13 @@ func tryConnRead(conn net.Conn, expectedRead string, timeout time.Duration) erro
 // used for connect/ack messages, not general purpose writes after the vsock
 // connection is established fully.
 func tryConnWrite(conn net.Conn, expectedWrite string, timeout time.Duration) error {
-	conn.SetDeadline(time.Now().Add(timeout))
-	defer conn.SetDeadline(time.Time{})
+	err := conn.SetDeadline(time.Now().Add(timeout))
+	if err != nil {
+		return err
+	}
+	defer func() {
+		_ = conn.SetDeadline(time.Time{})
```

Comment:
I really don't like wrapping these in defer statements to just ignore the error anyways. Prefer `defer conn.SetDeadline(time.Time{})` instead.

Reply:
Perhaps adding //nolint:errcheck above the defer may be preferable in cases where we are ignoring the error anyways.

Reply:
Let me return the err instead of ignoring.

Reply:
> Perhaps adding `//nolint:errcheck` above the defer may be preferable in cases where we are ignoring the error anyways.

Unless a linter provides a ton of value (and I'm not convinced that `errcheck` does) my general preference is to disable a faulty linter rather than trying to hold its hand.  There are a number of valid reasons to ignore a returned error and I think this type of linting is too aggressive.

Reply:
If so, I will exclude errcheck and just upgrade golangci-lint.